traitlets/traitlets.py

```diff
@@ -319,6 +320,23 @@ class that does the following:
        accept superclasses for :class:`This` values.
     """
 
+    def instance_init(self, obj):
```

Comment:
can you add a `name = None` class attribute here, as documentation that this will be assigned by the metaclass?